manifests/config.pp

```diff
@@ -49,9 +50,20 @@
       $_parameters = undef
     }
 
+    ###
+    # configure external tomcat datasource. See, for example:
+    # https://confluence.atlassian.com/doc/configuring-a-mysql-datasource-in-apache-tomcat-1867.html
+    ###
+    # Step 3. Configure Tomcat: Set the JDBC Resource for external database.
+
+    $jdbc_path = "${path}/Engine/Host/Context[#attribute/path='${context_path}']/Resource/#attribute"
+
+    if ! empty($tomcat_jdbc_settings) {
+      $_jdbc = suffix(prefix(join_keys_to_values($tomcat_jdbc_settings, " '"), "set ${jdbc_path}/"), "'")
+    }
     $_context_path_changes = "set ${path}/Engine/Host/Context/#attribute/path '${context_path}'"
 
-    $changes = delete_undef_values([$_parameters, $_context_path_changes])
+    $changes = delete_undef_values([$_parameters, $_context_path_changes, $_jdbc])
```

Comment:
Here, `$_jdbc` will be undefined (and thus fail with strict variables) if `$tomcat_jdbc_settings` is empty.

Perhaps set `$_jdbc` using a conditional expression or something?